compiler/fir/tree/src/org/jetbrains/kotlin/fir/expressions/impl/FirArgumentListWithResolutionInfo.kt

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright 2010-2021 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.fir.expressions.impl
+
+import org.jetbrains.kotlin.fir.declarations.FirValueParameter
+import org.jetbrains.kotlin.fir.expressions.FirExpression
+import org.jetbrains.kotlin.fir.resolve.substitution.ConeSubstitutor
+import org.jetbrains.kotlin.fir.types.ConeKotlinType
+
+interface FirArgumentListWithResolutionInfo {
```

Comment:
I'd just name this `FirArgumentList` right now. If for whatever reason you need to add more data here than "resolution info", it would avoid renaming in the future.

Reply:
I can't call it `FirArgumentList` because of https://github.com/JetBrains/kotlin/blob/9d557c8b796b0b377e3da48ae284e3e09c41f893/compiler/fir/tree/gen/org/jetbrains/kotlin/fir/expressions/FirArgumentList.kt#L18

Reply:
Oops. Ok I guess it's fine as-is.